docs/book/src/developer/tilt.md

```diff
@@ -192,6 +192,29 @@ COPY --from=tilt-helper /usr/bin/docker /usr/bin/docker
 COPY --from=tilt-helper /go/kubernetes/client/bin/kubectl /usr/bin/kubectl
 ```
 
+
+**extra_args** (Array, default=[]): An array of additional arguments to pass to the main binary configured
+for this provider. Each item in the array will be passed is to the manager. 
+
+Example:
+
+```json
+{
+    "name": "core",
+    "config": {
+        "context": ".",
+        "image": "gcr.io/k8s-staging-cluster-api/cluster-api-controller",
+        "extra_args": ["--feature-gates=MachinePool=true"]
```

Comment:
If I'm understanding this correct, in order to test a feature gate, you would have to override the necessary `tilt-provider.json` files with the correct `extra_args`. Is that correct?

I'm wondering if we can simplify the developer configuration a bit by giving a mapping of provider -> extra_args in `tilt-settings.json` instead?

Reply:
I'm a little unfamiliar with tilt -- is tilt-settings just an "us" thing and there's not any particular convention around using it? If so I definitely agree.

I waffled back and forth on this, it wasn't clear to me if tilt-settings was supposed to be like a driver for the specific invocation, or if it's more just a lib and we can do whatever we want. Sounds like we can go for it?

Reply:
It's an "us" thing, we created it as a way to add config for things like which downstream repos to consume and injecting things like the base64 encoded credentials for providers, etc